src/App.tsx

```diff
@@ -9,32 +9,53 @@ export const tabs = [
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
-export const App: React.FC = () => {
+interface TabType {
+  id: string;
+  title: string;
+  content: string;
+}
+
+type TabListProps = {
+  tab: TabType[],
+};
+
+export const App: React.FC<TabListProps> = () => {
+  const [activeTab, seActiveTab] = useState(tabs[0]);
+
+  const handleTabClick = (tab: TabType) => {
+    seActiveTab(tab);
+  };
+
   return (
     <div className="section">
       <h1 className="title">
-        Selected tab is Tab 1
+        Selected tab is
+        {` ${activeTab.title}`}
       </h1>
 
       <div data-cy="TabsComponent">
         <div className="tabs is-boxed">
           <ul>
-            <li className="is-active" data-cy="Tab">
-              <a href="#tab-1" data-cy="TabLink">Tab 1</a>
-            </li>
-
-            <li data-cy="Tab">
-              <a href="#tab-2" data-cy="TabLink">Tab 2</a>
-            </li>
-
-            <li data-cy="Tab">
-              <a href="#tab-3" data-cy="TabLink">Tab 3</a>
-            </li>
+            {tabs.map((tabItem) => (
+              <li
+                key={tabItem.id}
+                className={tabItem === activeTab ? 'is-active' : ''}
+                data-cy="Tab"
+              >
+                <a
+                  href={`#${tabItem.id}`}
```

Comment:
Using an onClick event on an anchor tag may cause the page to refresh, consider using a button or preventing the default behavior of the anchor tag.